kcidb/cache/__init__.py

```diff
@@ -0,0 +1,149 @@
+import hashlib
+import requests
+from email.header import Header
+import urllib.parse
+from google.cloud import storage
+
+class Client:
+    def __init__(self, bucket_name, max_store_size):
+        """
+        Initialize a cache client.
+        Args:
+            bucket_name:    The name of the GCS bucket containing the cache.
+            max_store_size: Maximum size the file can have to be stored.
+        """
+        self.bucket_name = bucket_name
+        self.max_store_size = max_store_size
+        self.client = storage.Client()
+
+    def _extract_content_disposition(self, response):
+        content_disposition = response.headers.get("Content-Disposition")
+
+        if content_disposition is None:
+            parsed_url = urllib.parse.urlparse(response.url)
+            original_filename = parsed_url.path.rsplit('/', 1)[-1]
+
+            encoded_filename = Header(original_filename, "utf-8").encode()
+
+            content_disposition = f'inline; filename="{encoded_filename}"'
```

Comment:
Let's switch to `attachment` disposition to be safe, here. We don't want browsers to try opening binary files directly. We can switch to using `inline` selectively for certain MIME types if that becomes a problem, later.

Reply:
done